map refresh after config change

maurizioandreotti opened the pull request:
Investigating why the mapkey value is not passed I found that the new map should also be reloaded if the config of the selected map is changed. but i wasnt able to find how to invoke the map window

wb8tyw commented on d_rats/mainapp.py:
```diff
@@ -1089,6 +1089,10 @@ def _refresh_map(self):
             mapurl = self.config.get("settings", "mapurlbase")
             mapkey = None
 
+        #todo: I think here it should trigger a refresh of the new map window after that a config reload/change is done
```
This will trigger at least one pylint issue because of the text "todo".

Pylint flags all "todo" or similar text like "fixme" as the pylint authors do not think they should ever be in production code.

As this is not currently feasible with D-rats, above that comment needs a line "# pylint: disable=fixme"

pylint may also trigger a messages about line length and the lack of space after the # character.

https://github.com/wb8tyw/D-Rats/wiki/Conversion-to-Python3-Notes#pylint

On the python3_tyw branch, I am trying to have no pylint/pylance issues active.

In general all changed modules should be scanned by the pylint program before they are pushed in a PR.

pylint should be available from the same source as the python interpreter for a distribution or it can be installed with pip from Pypi.

Some pylint issues flagged in d-rats can not be fixed and must be suppressed as above.
It is always best if possible to fix the code to be pass pylint if practical.
Currently I am trying to only add pylint suppression code to for GTK required issues, fixme, and complaints about modules having too many lines, variables, and other minor formatting issues. 

We can also tailor the pylintrc file to make some checks more or less strict for some of the issues.

pylint has found several existing bugs in d-rats as I have gone through the conversions.

maurizioandreotti commented:
thanks for you reply

in the original code I had added those lines to be able to change the map
without quitting and reopening d-rats-
i am not aware of piynt so pls feel free to

from the current implementation in py3 i am not sure to understand where it
invokes the map window, as in all places i looked the code seems to pass
both the mapurl and mapkey, but mapkey doesnt seem to arrive and be really
used. (i keep getting thunderforest maps with the watermark "api key
required")

thanks



Il giorno mar 1 feb 2022 alle ore 15:32 John E. Malmberg <
***@***.***> ha scritto:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In d_rats/mainapp.py
> <https://github.com/wb8tyw/D-Rats/pull/51#discussion_r796654013>:
>
> > @@ -1089,6 +1089,10 @@ def _refresh_map(self):
>              mapurl = self.config.get("settings", "mapurlbase")
>              mapkey = None
>
> +        #todo: I think here it should trigger a refresh of the new map window after that a config reload/change is done
>
> This will trigger at least one pylint issue because of the text "todo".
>
> Pylint flags all "todo" or similar text like "fixme" as the pylint authors
> do not think they should ever be in production code.
>
> As this is not currently feasible with D-rats, above that comment needs a
> line "# pylint: disable=fixme"
>
> pylint may also trigger a messages about line length and the lack of space
> after the # character.
>
> https://github.com/wb8tyw/D-Rats/wiki/Conversion-to-Python3-Notes#pylint
>
> On the python3_tyw branch, I am trying to have no pylint/pylance issues
> active.
>
> In general all changed modules should be scanned by the pylint program
> before they are pushed in a PR.
>
> pylint should be available from the same source as the python interpreter
> for a distribution or it can be installed with pip from Pypi.
>
> Some pylint issues flagged in d-rats can not be fixed and must be
> suppressed as above.
> It is always best if possible to fix the code to be pass pylint if
> practical.
> Currently I am trying to only add pylint suppression code to for GTK
> required issues, fixme, and complaints about modules having too many lines,
> variables, and other minor formatting issues.
>
> We can also tailor the pylintrc file to make some checks more or less
> strict for some of the issues.
>
> pylint has found several existing bugs in d-rats as I have gone through
> the conversions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wb8tyw/D-Rats/pull/51#pullrequestreview-869212063>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2W7OTT3HGSTK3RTDHWJ7TUY7VHPANCNFSM5NHD6Z4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

wb8tyw commented:
In the MapTile class, the map_url_key is added to the URI path in the remote_path method.

The Maptile class stores it as a class variable so it is common to all instances of the class.

The class method MapTile.set_map_info sets the the map url and key into the class.

The MapWindow class has a static method set_base_dir() that takes the map parameters.
This calls the MapTile.set_map_info method.

The mainapp calls MapWindow (Named Map.Window) to set the parameter when a map is selected in the _refresh map method.

I would recommend that you look up the pylint tool.  It is an essential tool that can find hidden bugs in a python program, in addition to being nit-picky about minor stuff.

maurizioandreotti commented:
thanks - i was asking as trying to investigate why thunderforest maps are
still displayed with the "no key watermark"
to do this i added a couple of logger lines

in MapTile modified remote_path method() to report in log what was received:

 def remote_path(self):
        '''
        Remote Path.

        :returns: URL of path
        :rtype: str
        '''

        #note by mao to track what is passed
        self.logger.info("remote_path: using map_url : %s", self._map_url )
        self.logger.info("remote_path: using map_url_key : %s",
self._map_url_key)

        remote_path = self._map_url + (self.path())
        if self._map_url_key:
            remote_path += self._map_url_key

        self.logger.info("remote_path: uri set to: %s", remote_path)
        return remote_path

        self.logger.info("_refresh_map: reconfigured mapkey to: %s", mapkey)



then i removed all tiles from map folder, just to force a fresh download
and run  Drats with debug option:

python3 ./d-rats.py --loglevel DEBUG
02/02/2022 10:01:16:INFO:D-Rats:main: re-config option found -- Reconfigure
D-rats
02/02/2022 10:01:18:INFO:Version:HTTP_CLIENT_HEADERS={'User-Agent':
'd-rats/0.4.00 pre-release 1'}
02/02/2022 10:01:27:INFO:DratsConfig:File /home/pi/.d-rats-ev/d-rats.config
02/02/2022 10:01:27:INFO:MainApp:_refresh_lang: Loading locale `en'
(...)
02/02/2022 10:01:39:INFO:MainApp:_refresh_map: reconfigured mapurl to:
http://tile.thunderforest.com/cycle/
02/02/2022 10:01:39:INFO:MainApp:_refresh_map: reconfigured mapkey to:
?apikey=5a1a4a79354244a38707d83969fd88a1
02/02/2022 10:01:39:INFO:SocketDataPath:getline: Host does not require
authentication

but then when i open the map window i see that the key has no value and
therefore it's not used to fetch:

(...)
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url :
http://tile.thunderforest.com/cycle/
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url_key : None
02/02/2022 10:02:15:DEBUG:MapTile:fetch: opened
http://tile.thunderforest.com/cycle/14/8616/5838.png
02/02/2022 10:02:15:DEBUG:MapTile:fetch: opened
http://tile.thunderforest.com/cycle/14/8616/5837.png
02/02/2022 10:02:15:INFO:MapTile:remote_path: uri set to:
http://tile.thunderforest.com/cycle/14/8616/5842.png
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url :
http://tile.thunderforest.com/cycle/
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url_key : None
02/02/2022 10:02:15:INFO:MapTile:remote_path: uri set to:
http://tile.thunderforest.com/cycle/14/8616/5843.png
02/02/2022 10:02:15:DEBUG:MapTile:fetch: opened
http://tile.thunderforest.com/cycle/14/8616/5840.png
02/02/2022 10:02:15:DEBUG:MapTile:fetch: opened
http://tile.thunderforest.com/cycle/14/8616/5839.png
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url :
http://tile.thunderforest.com/cycle/
02/02/2022 10:02:15:DEBUG:MapTile:fetch: opened
http://tile.thunderforest.com/cycle/14/8616/5841.png
02/02/2022 10:02:15:INFO:MapTile:remote_path: using map_url_key : None

wb8tyw commented:
The only thing that I can think of is that when I made the map_key an optional parameter by giving it a default value, it may have required the callers to use the variable name when calling it or the parameter is ignored.

such as "Map.Tile.set_map_info(dir, url, map_url_key=mapkey)"

Since all callers of these routines pass a url_key, there is no reason to have these methods have a default value.

At a previous employer though, the standard practice was always specify the parameter name in the call to a method as it avoids problems if the target method parameters change in the future.

Right now I am still trying to get points plotted in the right place on the map.
I think I am getting closer.
Have found existing bugs in the original source:
* __add__ method of the MapTile, needs to make sure that the resulting tile from the map is in the map bounds.  Bug is obvious at zoom level 2, but can show up at any zoom level if you set the map center certain locations.
* num2deg can fail with an Overflow error because some y positions can result sinh of infinity.
* Station position updates coming while the map is rendering stations causing GTK internal errors and sometimes segfaults crashes of d-rats.  Have not yet investigated a solution. 
* Sometimes the map at certain zoom levels, switches some tiles to bad tiles based on some event that may be related to mouse movements and clicks or activity on other windows.  Have not figured that one out yet either.  May be related to the the previous item.

wb8tyw commented:
These changes have been included in https://github.com/wb8tyw/D-Rats/pull/55.

The MapTile method for setting the new map information may be a better place to trigger the map to update to the new selected map.

wb8tyw commented:
Closing this PR as it has been included in another PR.
